src/Raven.Server/Utils/ThrottledManualResetEventSlim.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Threading;
+using Sparrow.Threading;
+
+namespace Raven.Server.Utils
+{
+    public class ThrottledManualResetEventSlim : IDisposable
+    {
+        private readonly ManualResetEventSlim _mre;
+        private readonly MultipleUseFlag _setCalled = new MultipleUseFlag();
+        private Timer _timer;
```

Comment:
TODO: Instead of creating separate timers for each `ThrottledManualResetEventSlim ` instance we might use pool of timers having same interval and just register there

Reply:
We already have TimeoutManager, no? Exactly for that.

Reply:
Initially I was thinking about creating something dedicated here - similar impl to what we have in `TimeoutManager`, but indeed it looks we can just spawn a task for that purpose and use `TimeoutManager` directly. Something like:

```
_timerTask = Task.Run(async () =>
{
	while (_token.IsCancellationRequested == false)
	{
		await TimeoutManager.WaitFor(_throttlingTimeInterval.Value, _token);

		if (_setCalled.Lower())
		{
			_mre.Set();
		}
	}
}, _token);
```